packages/bridge-ui-v2/src/components/Bridge/Amount.svelte

```diff
@@ -154,26 +154,42 @@
   // We want to debounce this function for input events.
   // Could happen as the user enters an amount
   const debouncedValidateAmount = debounce(validateAmount, 300);
+  let sanitizedValue = '';
 
-  // Will trigger on input events. We update the entered amount
-  // and check it's validity
   function inputAmount(event: Event) {
+    invalidInput = false;
+    $validatingAmount = true; // During validation, we disable all the actions
     if (!$selectedToken) return;
+    const target = event.target as HTMLInputElement;
+    let value = target.value;
 
-    const { value } = event.target as HTMLInputElement;
-    if (
-      $selectedToken.type === TokenType.ERC721 ||
-      $selectedToken.type === TokenType.ERC1155 ||
-      !$selectedToken.decimals
-    ) {
-      $enteredAmount = BigInt(value);
+    if ($selectedToken.type === TokenType.ERC1155) {
+      // For ERC1155, no decimals are allowed
+      if (value.includes('.') || value.includes(',')) {
```

Comment:
`if (/[.,]/.test(value)) {`
I think this is better.

Reply:
Essentially comes down to readability, but as regex is a bit more performant and it is happening on typing input, I agree! Thanks